jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/test/java/org/jbpm/workbench/es/client/editors/requestlist/RequestListViewImplTest.java

```diff
@@ -181,5 +196,65 @@ public void initDefaultFiltersOwnTaskFilter() {
         verify(presenter).setAddingDefaultFilters(true);
     }
+    
+    @Test
+    public void multiStateActionHasCellTest(){
+        RequestListViewImpl.ActionHasCell multiStateCell = new RequestListViewImpl.ActionHasCell("", cellDelegate,
+                RequestStatus.QUEUED,
+                RequestStatus.DONE,
+                RequestStatus.CANCELLED
+        );
+        for(RequestStatus rs: RequestStatus.values()){
+            RequestSummary testRequest = new RequestSummary(1L, new Date(), rs.toString(), null,
+                    "Test message", "Test key", 1, 1, "procName", 55L, "procInstDesc");
+            boolean shouldRender = false;
+            switch(rs){
+                case QUEUED:
+                case DONE:
+                case CANCELLED:
+                    shouldRender = true;
+                    break;
+            }
+            runActionHasCellTest(multiStateCell, testRequest, shouldRender);
+        }
+    }
+    
+    @Test
+    public void singleStateActionHasCellTest(){
+        for(RequestStatus cellRs: RequestStatus.values()){
+            RequestListViewImpl.ActionHasCell multiStateCell = new RequestListViewImpl.ActionHasCell("", cellDelegate, cellRs);
+            for(RequestStatus valueRs: RequestStatus.values()){
+                RequestSummary testRequest = new RequestSummary(1L, new Date(), valueRs.toString(), null,
+                        "Test message", "Test key", 1, 1, "procName", 55L, "procInstDesc");
+                boolean shouldRender = (valueRs == cellRs);
+                runActionHasCellTest(multiStateCell, testRequest, shouldRender);
+            }
+        }
+    }
+    
+    @Test
+    public void predicateActionHasCellTest(){
+        RequestSummary testRequest = new RequestSummary(1L, new Date(), "DONE", null, "Test message", "Test key", 1, 1, "procName", 55L, "procInstDesc");
+
+        RequestListViewImpl.ActionHasCell alwaysDisplayedActionCell = new RequestListViewImpl.ActionHasCell("", (val -> true), cellDelegate);
+        runActionHasCellTest(alwaysDisplayedActionCell, testRequest, true);
+
+        RequestListViewImpl.ActionHasCell neverDisplayedActionCell = new RequestListViewImpl.ActionHasCell("", (val -> false), cellDelegate);
+        runActionHasCellTest(neverDisplayedActionCell, testRequest, false);
+    }
+    
+    private void runActionHasCellTest(RequestListViewImpl.ActionHasCell cell, RequestSummary val, boolean shouldRender){
+        RequestListViewImpl.ActionHasCell cellMock = spy(cell);
+        SafeHtmlBuilder cellHtmlBuilder = mock(SafeHtmlBuilder.class);
+        doAnswer(invocationOnMock -> {
+            invocationOnMock.callRealMethod();
+            verify(cellHtmlBuilder, times(shouldRender ? 1 : 0)).append(any());
+            return null;
+        }).when(cellMock).render(any(), any(), eq(cellHtmlBuilder));
+
+        cellMock.render(cellContext, val, cellHtmlBuilder);
+
+        verify(cellMock).render(cellContext, val, cellHtmlBuilder);
```

Comment:
@bsremac I think both implementations are valid, though yours is shorter. I'll replace it.